modules/navier_stokes/src/fvkernels/INSFVTurbulentAdvection.C

```diff
@@ -53,7 +57,10 @@ INSFVTurbulentAdvection::computeQpResidual()
                                       limiterType(_advected_interp_method),
                                       MetaPhysicL::raw_value(v) * _normal > 0),
                              determineState());
-  return _normal * MetaPhysicL::raw_value(v) * rho_face * var_face;
+  if (_newton_solve)
+    return _normal * v * rho_face * var_face;
```

Comment:
I think here it s more about keeping the derivatives (which possibly improved out Newton convergence, but I did not really see it get better) than preserving the sparsity pattern

Reply:
>  What would your opinion be on separating kernels and materials for segregated solves and Newton solves?

This is where we are going in a way already. But we need to contribute the linearFV version of these kernels.
Otherwise wrt the nonlinear kernel, no I dont want to duplicate roughly the same thing. Especially since I am trying to not have fvkernels be user facing anymore with these physics

Reply:
How would having duplicated kernels for nonlinear FV Newton and SIMPLE be easier to maintain?

Reply:
Yeah I think when I wrote this down I didn't consider that hopefully the segregated functionalities will migrate to linearFV and then we'll have no trouble maintaining shared kernels between two very different solution algorithms.